### DISCUSSION TITLE
useEnsAvatar. Is it possible to pass the addresOrName as a prop?

### DISCUSSION THREAD

**Discussion by Javier-Szyfer** [wevm/wagmi#95] [Q&A] [answered]
Hi devs! Can i get some help with the following:

I would like to call the getEnsAvatar method from inside a  function and pass the the addressOrName as props.
Im doing someting wrong here because i got this _Error : addressOrName is required_

For example `const avatar = async (address) => {
const img = await getEnsAvatar(address)
}
`

```
  const [{ data, error, loading }, getEnsAvatar] = useEnsAvatar({
    skip: true,
  })
```

Thanks!

**Comment by tmm** [comment, marked as answer]
[`getEnsAvatar`](https://wagmi-xyz.vercel.app/docs/hooks/useEnsAvatar#getensavatar) required an object:

```
getEnsAvatar({
  addressOrName: address,
})
```

**Reply by Javier-Szyfer** [reply]
upss you're right : )
